File: src/main/scala/model/Department.scala
package model

import model.SchoolId.SchoolId
import core.StringUtilities._
import scala.slick.driver.MySQLDriver.simple._

case class Department(
  schoolId: SchoolId,
  schoolSpecificId: String,
  name: String
) {
  def primaryKey: String = {
    s"${schoolId}_$schoolSpecificId".toSafeId
  }
}

class Departments(tag: Tag) extends Table[Department](tag, "departments") {
  def departmentId = column[String]("department_id", O.PrimaryKey)
  def schoolId = column[Int]("school_id")
  def schoolSpecificId = column[String]("school_specific_id")
  def name = column[String]("name")

  def schoolDerivedIndex = index("dep_school_derived_index", (schoolId, schoolSpecificId), unique = true)
  def school = foreignKey("dep_school_fk", schoolId, TableQuery[Schools])(_.schoolId)

  def * = (departmentId, schoolId, schoolSpecificId, name) <> (createDepartment, createTuple)

  private def createDepartment(tuple: (String, Int, String, String)): Department = {
    Department(SchoolId(tuple._2), tuple._3, tuple._4)
  }

  private def createTuple(department: Department): Option[(String, Int, String, String)] = {
    Option((department.primaryKey, department.schoolId.id, department.schoolSpecificId, department.name))
  }
}

File: server/code/model/department.py
from base import Base
from sqlalchemy import Column, String, Enum, ForeignKey, Integeer

class Department(Base, Ninja_Model_Mixin):
    __tablename__ = 'departments'
    department_id = Column('department_id', String(254), primary_key = True)
    school_id = Column('department_id', Integer, ForeignKey("schools.school_id"), nullable = False)
    school_specific_id = Column('school_specific_id', String(254), nullable = False)
    name = Column('name', String(254), nullable = False)
